### ISSUE TITLE
Multi-Language Scripting API

### ISSUE THREAD

**Issue by erherhh4herh** [PanosK92/Directus3D#11]
I know this engine is set to support C# scripting soon, but I have an idea that could potentially enable the engine to support a multitude of programming languages while only maintaining one. You may have heard of a programming language called "Haxe". Well, not many seem to know that it's compiler has an interesting feature that allows it to trans-pile to multiple different languages, including C#, C++, lua, java, etc! It might not be perfect, but the Haxe, including the compiler, is open-source... It's just an idea to consider!

If you wish to read up on the language and it's compiler, they have a manual:
https://haxe.org/manual/compiler-usage.html

Thanks for you time,
erherhh4herh.

**Comment by PanosK92** [comment]
I haven't heard of it actually. At first glance, the fact that it's open source and that it can generate source code for a multitude of languages are two big pluses. If Haxe hides the trans-pilation complexity well enough (e.g. it trully comes down to those compiler arguments) then I can't think of anything negative.

Of course, I'm just thinking out loud here, I will have a look at it during the weekend to get a more in-depth understanding. I'll leave this issue open as a reminder that scripting has to be taken care of at some point and/or further discussion etc.

Thanks for your time as well erherhh4herh :-)

**Comment by bayganik** [comment]
is C# scripting still continuing?

**Comment by PanosK92** [comment]
Yes. Although I'll probably switch to .Net Core instead of moving forward with Mono.
